### PR TITLE
Drive train update

### PR DESCRIPTION
**Pull request by emilesavoie**
Modifs:
- modif acc curve to non linear
- fix tank turn
- modif comm arb to can

### CODE REVIEW

File: rover_drive_train/src/teleop.cpp [hunk 1]
```diff
@@ -82,8 +82,8 @@ class Teleop : public rclcpp::Node
 
             if(_modeTankAngularInput != 0.0f) 
             {
-                speedLeftMotor += _modeTankAngularInput * speedFactor;
-                speedRightMotor -= _modeTankAngularInput * speedFactor;
+                speedLeftMotor += -1 * _modeTankAngularInput * speedFactor;
+                speedRightMotor -= -1 * _modeTankAngularInput * speedFactor;
```
Comment (Pmichaud29) [inline rover_drive_train/src/teleop.cpp]: -1.0f

File: rover_joy/src/joy_formator.cpp [hunk 2]
```diff
@@ -391,3 +392,10 @@ void JoyFormator::customStepsLogitech(rover_msgs::msg::Joy *formatted_joy)
         *formatted_joy = rover_msgs::msg::Joy();
     }
 }
+
+float JoyFormator::tanhVal(float _value)
+{
+    //input 0.8\arctan\left(3x\right) at https://www.desmos.com/calculator to view curve
+    float output = 0.8*atan(3*_value);
```
Comment (Pmichaud29) [inline rover_joy/src/joy_formator.cpp]: 3.0f

File: rover_joy/src/joy_formator.cpp [hunk 3]
```diff
@@ -391,3 +392,10 @@ void JoyFormator::customStepsLogitech(rover_msgs::msg::Joy *formatted_joy)
         *formatted_joy = rover_msgs::msg::Joy();
     }
 }
+
+float JoyFormator::tanhVal(float _value)
```
Comment (Pmichaud29) [inline rover_joy/src/joy_formator.cpp]: float value_

File: rover_msgs/include/rovus_lib/macros.h [hunk 4]
```diff
@@ -31,8 +31,12 @@
 #define IN_ERROR(VAR, ERROR, GOAL) \
     ((abs(VAR) < (abs(GOAL) + ERROR) && abs(VAR) > (abs(GOAL) - ERROR)))
 
-#define MAP(result_type, value, from_min, from_max, to_min, to_max) \
-    (result_type)(((float)value - (float)from_min) / ((float)from_max - (float)from_min)) * ((float)to_max - (float)to_min) + (float)to_min
+#define MAP(RESULT_TYPE, VALUE, FROM_MIN, FROM_MAX, TO_MIN, TO_MAX) \
+    (RESULT_TYPE)(((float)VALUE - (float)FROM_MIN) / ((float)FROM_MAX - (float)FROM_MIN)) * ((float)TO_MAX - (float)TO_MIN) + (float)TO_MIN
+
+//Returns a value between set constrains
+#define CONSTRAIN(VALUE, UPPER_RANGE, LOWER_RANGE) \
```
Comment (Pmichaud29) [inline rover_msgs/include/rovus_lib/macros.h]: value, lower, pis higher jpense ça serait plus intuitif

**Review by Pmichaud29 — CHANGES_REQUESTED** [review]

**Comment by emilesavoie** [comment]
modif done

**Review by Pmichaud29 — APPROVED** [review]

**Pull request merged**